### ISSUE TITLE
Using Bulbs with Remote Only? - 601rcw

### ISSUE THREAD

**Issue by gotminer** [bren1818/TCPLightingWebInterface#17]
Is there a way to connect bulbs to a zone on the remote and eliminate the gateway altogether? I saw someone mention it, but I never found any instructions.

I've been using my system with the remote only since the app stopped working, but I recently had a power outage at my house and it caused more than half of my bulbs to reset. The remote still works with one set of bulbs in one lighting fixture, but the other three fixtures just flash when I flip the light switch on the wall. At this point my preference would be to pair the bulbs with my remote and not use the gateway at all.

I've attempted to use the web interface with XAMPP, but I'm still getting a connection refused error when I enter the IP into a browser. Not sure where to go from here in terms of the web interface. I edited the config.inc.php file with the IP address of the gateway, pressed the sync gateway button on the gateway, but always get a connection refused error in my browser.

**Comment by bren1818** [comment]
Two things to check.

1. Make sure the machine you're running the app on doesn't have a firewall
blocking traffic . (If your connecting not by localhost or on another
device).

2. If you're running the new xampp with php 7, you may need to uncomment
line 39 in the include.php to specify a cipher.

The app should have pretty much all the functionality to create rooms, add
or remove bulbs etc.

I recommend setting up a static ip for your bridge on your router so it
doesn't change. You will need the tcp bridge since it is what connects to
the lights.

You -> TCP App on webserver-> Bridge->Lights

One other thing you can try is delete the .htaccess file and see if it
loads up then.

I'm AFK currently but can help more Monday if you can't get it to go

**Comment by gotminer** [comment]
I have tested with Windows Firewall Disabled and Enabled. There are no other firewalls on this machine. I can ping the gateway in both instances.

I am running XAMPP version 5.6.40.

I have deleted the .htaccess file.

Can the bulbs be put in discovery mode or whatever you call it when you flip the switch on-off-on-off-on-off-on to make them flash and then paired to a zone on the remote without using the gateway and web interface?

![1](https://user-images.githubusercontent.com/44104559/72669664-be94f480-3a02-11ea-8662-2a8d4ac4e03a.JPG)

![2](https://user-images.githubusercontent.com/44104559/72669666-c2287b80-3a02-11ea-9853-2863ca13af75.JPG)

![3](https://user-images.githubusercontent.com/44104559/72669669-c48ad580-3a02-11ea-8ff9-55f0f149d5b4.JPG)

**Comment by bren1818** [comment]
Ok so looking at your screen shot, you connect to the webserver ip in the
browser, not your bridge ip. Try 127.0.0.1 or localhost or the ip of your
computer in your browser. (You can figure that one out by opening command
prompt and typing ipconfig and looking for the ipv4 address) Assuming xampp
is running it should give you the landing page.

You need either the webapp or one of the mobile apps to program the lights
/ put them in rooms / groups. I believe I did include pdfs of the original
product documentation with this project to help reset the bulbs if you need
to.

**Comment by gotminer** [comment]
Thanks so much for your help! I'm getting further than I was before. Apparently VMWare was using port 443 and preventing Apache from running. That part is working now, but when I type localhost or the ip of my machine into a browser I get the attached message. 

What does it mean by "paste your token in the config.inc.php."? I don't see any instructions for this part.

![4](https://user-images.githubusercontent.com/44104559/72671963-9ddb9780-3a20-11ea-9bdf-5a78bde425d9.JPG)

**Comment by gotminer** [comment]
I do see where to paste the token, but I guess my question is where do I get the token? The wiki says to paste the code that is generated during the gateway sync, but I'm not sure where I'm supposed to see that code.

![5](https://user-images.githubusercontent.com/44104559/72672034-00816300-3a22-11ea-8fd2-33815b528357.JPG)

**Comment by bren1818** [comment]
Assuming you have the ip correct for your bridge and you've pressed the
sync button on the bridge, when you refresh the page it should grab and
save the token automatically. The pasting is for an older setup.

You may have to uncomment (remove the //) on the line i mentioned earlier
if it isn't getting the token. Also make sure that the php. Ini has curl
enabled but xampp should have it correct by default.

**Comment by gotminer** [comment]
Hmmm. I'm positive the bridge ip is correct. 

I did uncomment line 39 in the include.php file and checked the php.ini file. It appears to have curl enabled. Still having the same issue though.

![6](https://user-images.githubusercontent.com/44104559/72672314-29a3f280-3a26-11ea-98aa-e08b4e52be55.JPG)

![7](https://user-images.githubusercontent.com/44104559/72672315-2b6db600-3a26-11ea-9209-7c84654ebf44.JPG)

![8](https://user-images.githubusercontent.com/44104559/72672316-2dd01000-3a26-11ea-8f5e-5070d4b2e7bc.JPG)

![9](https://user-images.githubusercontent.com/44104559/72672318-2f99d380-3a26-11ea-81dd-ee9388094b24.JPG)

**Comment by bren1818** [comment]
Everything looks right. You had mentioned this was running on a vm?

Are you able to access the bridge ip via https or ping it from command
prompt?

**Comment by sktaylortrash** [comment]
If this is XAMPP for Linux the web server doesn't have write permissions to the html directory by default. So it can't create tcp.token automatically.  I've always just recommended creating the file manually and using chmod to set the permissions to 766 This should let the automatic token creation work properly

**Comment by gotminer** [comment]
It's not running on a vm. I just had VMWare workstation installed on this computer for something else and it was using port 443 for something. After I removed the software, Apache connected without an issue.

Yes I can ping the ip of the bridge from this pc. Can't access via https though.

It's XAMPP for Windows. I did give "everyone" full control of c:\xampp.

**Comment by sktaylortrash** [comment]
Odd, but it still couldn't hurt to try making a blank file called tcp.token

**Comment by bren1818** [comment]
Odd. So in theory, if you go to https://{your bridge ip} you should get an
invalid cert warning and then it resolve to a 403.

The lighting web app doesn’t necessarily need to be run on 443 (https) you
can also access it on http

**Comment by gotminer** [comment]
Yes, Bren. When I try to access the bridge ip via https, I get invalid cert and then a 403.

So I changed the lighting bridge port from 443 to 80 in the config.inc.php file and I get this ...

![10](https://user-images.githubusercontent.com/44104559/72672977-67f2df00-3a31-11ea-84ec-2b41add95dd2.JPG)

And then if I create a tcp.token file in htdocs folder, I get this ...

![11](https://user-images.githubusercontent.com/44104559/72672983-7e993600-3a31-11ea-9b91-beb7d93de425.JPG)

**Comment by bren1818** [comment]
Really based on what you described, everything seems correct. The only two
lines you would/should need to update are:

define("LIGTHING_BRIDGE_IP", "10.0.0.179"); // IP address of TCP
Bridge/Gateway
define("LIGHTING_BRIDGE_PORT",

define("LOCAL_URL", http://xamp-server-ip

Then once you refresh the page after tapping the sync button everything
should be good. If not, the only thing I can think is perhaps the newer
version of xampp is doing something odd or theres an update with php
preventing it from writing the token.

I'll try downloading it and setting it up from scratch tomorrow when I'm in
front of my desktop.

There may be logs you can check in an xampp sub directory

**Comment by gotminer** [comment]
What version of xampp would you suggest I go back to? I'm currently on 5.6.40.

**Comment by bren1818** [comment]
I just downloaded 7.4.1 (latest) version of xammp. Did the standard install
but skipped perl, sendmail.

Then stated xampp.

I then Deployed the application into the htdocs folder. Edited the
config.inc setting my bridge ip to my bridge ip and the local url to the ip
address of my server.

I navigated to the page and got a 500 telling me the .htaccess was
problematic so I deleted that.

Refreshed the page and it prompts that it needs the token from the bridge,
so I hit the sync button on the bridge and refreshed the page and viola,
worked ok.

So perhaps download the latest xampp and try again. I only needed to edit
two lines of the config.inc and removed the htaccess. No other changes were
needed.

Good luck!

**Comment by gotminer** [comment]
So I just performed the exact same thing on a machine with a fresh install of Win10 and disabled the firewall.

Standard install of xampp 7.4.1 minus perl and sendmail.

Started xampp. Apache is the only thing running in the xampp control panel.

Deployed application to htdocs folder.

Edited the config.inc file with only what you mentioned above.

Navigated to the page and got the 500 error.

Deleted htaccess, refreshed page, and got the "if you are seeing this, you haven't generated your token yet" error.

Pressed the sync button on the bridge and refreshed page. Same token error.

This is my bridge. Power light on. Lan light off. Wan light red. Lan light is off but bridge ip is still pingable. Anything out of the ordinary here? Anything in my router config that could be causing a problem?

![bridge](https://user-images.githubusercontent.com/44104559/72688051-c40d4000-3ad1-11ea-88d8-d28b67d3e4ed.jpg)

**Comment by bren1818** [comment]
It's possible you need to restart the bridge, they can be fussy. Pull the
power plug, let it sit off for a while and start it back up but otherwise I
see no reason for you to have an issue unless your router has a built in
firewall preventing your bridge to communicate which would be really odd.

Your LED indicator seems right. Mine has Wan lit up red, power illuminated
and lan off too.


All I can suggest really is see if you can assign an ip to the bridge
versus using dhcp (if you haven't already)  this may give you a bit more
control. Another option you can try is if you have another router , plug
just your pc and the bridge into it. You'd need to update your config
accordingly though for the new ips.

Not sure how much help I can be beyond this since each router is a bit
different.

**Comment by gotminer** [comment]
I'll leave it unplugged for several hours and see if that does anything. I've restarted it a zillion times, but only leaving it unplugged for a couple of seconds. 

How do I set a static ip on the bridge? I can't get to a web interface when putting the bridge ip into a browser.

I have a friend who has a spare bridge that I'm going to test tomorrow.

If all else fails, I'll test with a different router.

Thanks for all of your help!

**Comment by bren1818** [comment]
I set the static ip by assigning a specific ip address to a specific MAC
address (configured on my router). I try to statically assign as many
devices as possible in my network as it makes it easier for me to identify
them.

I hope the spare bridge assists. So long as the two devices are on the same
network it should work. I'm thinking there must be something on your
network preventing the cross communication.


Keep us posted,

Good luck

**Comment by gotminer** [comment]
I tested it on a different router and tried my friends gateway last week, still had the same problems. 

Gave up on it for a few days and then was reading another thread where someone was powering on the gateway with the sync button held in. What exactly does that do? I tried it and ended up with a ip, but the wan light would never turn solid red like normal. And the sync button didn't work at that point. After it sat there for awhile, I rebooted it again. It booted up like normal and the wan light turned solid red. 

So then I checked the webpage, got the sync message. Hit the sync button on the gateway, refreshed the page, and it finally worked! I can see my lighting groups and after it found all of the bulbs again, it knew which bulbs belonged to which group.

Whatever the gateway did when I powered it on while holding the sync button was what it needed. I'm really curious to know what it was.

**Comment by bren1818** [comment]
It sounds like your bridge might have been flaking out or had static built
up on it. Usually having it unplugged a short while resolves this.

I believe holding the sync button on boot does a mini reset, but I would
have to read the original documents. Holding the bottom may have even been
a placebo.  Im glad it worked for you though, Hopefully no further problems
and I'll keep your trick in mind in the event any others run into the same
issue.

Cheers :)

**Comment by TDBatt** [comment]
Yes you can pair the bulbs directly to a remote. I bought a set that was programmed like that, a remote and two bulbs. I reprogrammed them to my gateway.
